ZenPacks/zenoss/ZenPackLib/lib/spec/ZenPackSpec.py

```diff
@@ -466,6 +482,25 @@ def create_device_js_snippet(self):
         return self.create_js_snippet(
             'device', snippet, classes=device_classes)
 
+    def create_add_js_snippet(self):
+        """Register component add JavaScript snippet."""
+        device_classes = [
```

Comment:
Can you take a look at the check I put in there and see if it makes sense?

Also, I restricted the "user_defined" stuff to components only, and the menu items only get added if the to-one is a device (since the js is per-device class).

Meaning this feature isn't available for and won't work for a component 1:MC addable_component, only for a device 1:MC addable_component.